classes/external/enqueue_merge_request.php

```diff
@@ -0,0 +1,120 @@
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+/**
+ * Version information
+ *
+ * @package     tool
+ * @subpackage  mergeusers
+ * @author      Nicola Vallinoto, Liguria Digitale
+ * @author      Jordi Pujol-Ahulló, SREd, Universitat Rovira i Virgili
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+namespace tool_mergeusers\external;
+defined('MOODLE_INTERNAL') || die();
+global $CFG;
+require_once("{$CFG->libdir}/externallib.php");
+defined('MOODLE_INTERNAL') || die();
+use \tool_mergeusers\merge_request;
+use external_function_parameters;
+use external_multiple_structure;
+use external_single_structure;
+use external_value;
+class enqueue_merge_request extends \external_api {
+    /**
+     * Webservice enqueue_merge_request parameters
+     *
+     * @return external_function_parameters
+     */
+    public static function execute_parameters(): external_function_parameters {
+            return new external_function_parameters(
+            [
+                'removeuserfield' => new external_value(PARAM_TEXT, 'Remove user field'),
+                'removeuservalue' => new external_value(PARAM_RAW, 'Remove user value'),
+                'keepuserfield' => new external_value(PARAM_TEXT, 'Keep user field'),
+                'keepuservalue' => new external_value(PARAM_RAW, 'Keep user value'),
+            ]
+        );
+    }
+    /**
+     * Return identifier of the enqueue merge request
+     *
+     * @param $removeuserid
+     * @param $removeuservalue
+     * @param $keepuserid
+     * @param $keepuservalue
+     */
+    public static function execute(string $removeuserfield,
+                                     string $removeuservalue,
+                                   string $keepuserfield,
+                                   string $keepuservalue): int {
+        // Validate all of the parameters.
+        $params = self::validate_parameters(self::execute_parameters(),
+                                            ['removeuserfield' => $removeuserfield,
+                                            'removeuservalue' => $removeuservalue,
+                                            'keepuserfield' => $keepuserfield,
+                                            'keepuservalue' => $keepuservalue
+                                            ]);
+        global $DB;
+        // Insert of the merge request into mdl_merge_request_queue Moodle table.
```

Comment:
you should omit the prefix part of the table names like `mdl_` when referencing tables. Just use `merge_request_queue`. `mdl_` prefix is the default value, but it is not every where in use.

Reply:
done. the name of the table is: tool_mergeusers_queue